pkg/rancher-desktop/main/diagnostics/pathManagement.ts

```diff
@@ -0,0 +1,60 @@
+import { DiagnosticsCategory, DiagnosticsChecker, DiagnosticsCheckerResult, DiagnosticsCheckerSingleResult } from './types';
+
+import { ErrorHasExtendedAttributes, ErrorNotRegularFile, ErrorWritingFile } from '@pkg/integrations/manageLinesInFile';
+import mainEvents from '@pkg/main/mainEvents';
+import Logging from '@pkg/utils/logging';
+
+const console = Logging.diagnostics;
+const cachedResults: Record<string, DiagnosticsCheckerResult> = {};
+
+const CheckPathManagement: DiagnosticsChecker = {
+  id:       'PATH_MANAGEMENT',
+  category: DiagnosticsCategory.Utilities,
+  applicable() {
+    return Promise.resolve(['darwin', 'linux'].includes(process.platform));
+  },
+  check(): Promise<DiagnosticsCheckerSingleResult[]> {
+    return Promise.resolve(Object.entries(cachedResults).map(([id, result]) => {
+      return ({
+        ...result,
+        id,
+      });
+    }));
+  },
+};
+
+mainEvents.on('diagnostics-event', (id, state) => {
+  console.log('diagnostics-event', id, state);
+  if (id !== 'path-management') {
```

Comment:
nit; I noticed that `path-management` id is scattered all over the place, even in all the calls (around 4 calls) to `mainEvents.emit('diagnostics-event', 'path-management', { fileName, error });`.

Reply:
I'll make the `mainEvents.emit()` call typed so that a typo is a compile-time error.